tests/sentry/integrations/test_metric_alerts.py

```diff
@@ -0,0 +1,167 @@
+        """
+        alert_rule = self.create_alert_rule()
+        date_started = self.now
+        incident = self.create_incident(
+            self.organization,
+            title="Incident #1",
+            projects=[self.project],
+            alert_rule=alert_rule,
+            status=IncidentStatus.CLOSED.value,
+            date_started=date_started,
+        )
+        trigger = self.create_alert_rule_trigger(alert_rule, CRITICAL_TRIGGER_LABEL, 100)
+        self.create_alert_rule_trigger_action(
+            alert_rule_trigger=trigger, triggered_for_incident=incident
+        )
+        metric_value = 123
+        referrer = "metric_alert_custom"
+        notification_uuid = str(uuid.uuid4())
+
+        detector = self.create_detector(project=self.project)
+
+        open_period = (
+            GroupOpenPeriod.objects.filter(group=self.group).order_by("-date_started").first()
+        )
+        assert open_period is not None
+
+        metric_issue_context = MetricIssueContext.from_legacy_models(
+            incident, IncidentStatus.CLOSED, metric_value
+        )
+        # Setting the open period identifier to the open period id, since we are testing the lookup
+        metric_issue_context.open_period_identifier = open_period.id
+        metric_issue_context.group = self.group
+        assert metric_issue_context.group is not None
+
+        # XXX: for convenience, we populate the AlertContext with alert rule/incident information. In this test,
+        # we're just interested in how the method handles missing AlertRuleDetectors/IGOPs.
+        data = incident_attachment_info(
+            organization=incident.organization,
+            alert_context=AlertContext(
+                name=alert_rule.name,
+                # Setting the action identifier id to the detector id since that's what the NOA does
+                action_identifier_id=detector.id,
+                threshold_type=AlertRuleThresholdType(alert_rule.threshold_type),
+                detection_type=AlertRuleDetectionType(alert_rule.detection_type),
+                comparison_delta=alert_rule.comparison_delta,
+                sensitivity=alert_rule.sensitivity,
+                resolve_threshold=alert_rule.resolve_threshold,
+                alert_threshold=None,
+            ),
+            metric_issue_context=metric_issue_context,
+            notification_uuid=notification_uuid,
+            referrer=referrer,
+        )
+
+        fake_alert_rule_id = get_fake_id_from_object_id(detector.id)
+        fake_incident_identifier = get_fake_id_from_object_id(open_period.id)
+
+        # Build the link using the fake alert_rule_id and the fake incident identifier
+        assert (
+            data["title_link"]
+            == f"http://testserver/organizations/baz/alerts/rules/details/{fake_alert_rule_id}/?alert={fake_incident_identifier}&referrer={referrer}&detection_type=static&notification_uuid={notification_uuid}"
+        )
```

Comment:
Forgive me if I missed some context, but will users actually see these links? They won't work, right? Or will we be sending different links for single written detectors that work?